src/roles/factotum.js

```diff
@@ -72,6 +72,17 @@ class Factotum extends MetaRole {
         } else {
           this.goHome(creep)
         }
+        return
+      } else {
+        const tombstones = suicideBooth.lookFor(LOOK_TOMBSTONES).filter((t) => _.sum(t.store) > 0)
+        if (tombstones.length > 0) {
+          if (creep.pos.isNearTo(suicideBooth)) {
+            creep.withdraw(tombstones[0], Object.keys(tombstones[0].store)[0])
```

Comment:
Since RESOURCE_ENERGY is always defined, even when empty, this will get stuck withdrawing energy and will not be able to withdraw anything else.

Two options to resolve this are to either shuffle the results from Object.keys (so RESOURCE_ENERGY won't always be on top), or to filter it so RESOURCE_ENERGY is removed if it does not actually contain energy.

Reply:
Oh, yeah... I'll try to fix that in another PR.